From 7b74cc9adb77f025190557e47f6dadae110e6ab5 Mon Sep 17 00:00:00 2001
From: Colleen Murphy <colleen.murphy@suse.de>
Date: Tue, 5 Mar 2019 10:00:46 +0100
Subject: [PATCH] Fix rate semaphore for keystoneclient

When using keystoneclient sessions, the new parameter is not available
and breaks the keystoneclient unit tests[1]. Only use the semaphore
kwarg when using keystoneauth sessions.

[1] https://review.openstack.org/640953

Change-Id: I0cc7f2514e143ec532d8fb895618f7cf1fea9cc3
---
 keystoneauth1/adapter.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/keystoneauth1/adapter.py b/keystoneauth1/adapter.py
index a15fd54..6bba62f 100644
--- a/keystoneauth1/adapter.py
+++ b/keystoneauth1/adapter.py
@@ -217,6 +217,8 @@ class Adapter(object):
                 kwargs.setdefault('client_name', self.client_name)
             if self.client_version:
                 kwargs.setdefault('client_version', self.client_version)
+            if self._rate_semaphore:
+                kwargs.setdefault('rate_semaphore', self._rate_semaphore)
 
         else:
             warnings.warn('Using keystoneclient sessions has been deprecated. '
@@ -232,8 +234,6 @@ class Adapter(object):
         if self.raise_exc is not None:
             kwargs.setdefault('raise_exc', self.raise_exc)
 
-        kwargs.setdefault('rate_semaphore', self._rate_semaphore)
-
         return self.session.request(url, method, **kwargs)
 
     def get_token(self, auth=None):
-- 
2.21.0

